src/tool/hpcrun/sample-sources/amd.c

```diff
@@ -169,22 +171,17 @@ METHOD_FN(finalize_event_list)
   char* evlist = METHOD_CALL(self, get_event_str);
   char* event = start_tok(evlist);
 #endif
-  roctracer_init();
-
-  // Register flush function to turn off roctracer and flush traces 
-  // NOTE: this is a registered as a flush callback because is MUST precede 
-  //       GPU trace finalization, which is registered as a shutdown callback
-  device_finalizer_flush.fn = roctracer_fini;
-  device_finalizer_register(device_finalizer_type_flush, 
-                            &device_finalizer_flush);
-
-  // initialize gpu tracing 
-  gpu_trace_init();
-
-  // Register shutdown function to finalize gpu tracing and write trace files
-  device_trace_finalizer_shutdown.fn = gpu_trace_fini;
-  device_finalizer_register(device_finalizer_type_shutdown, 
-                            &device_trace_finalizer_shutdown);
+    roctracer_init();
+
+    // Init records
+    gpu_trace_init();
+
```

Comment:
Currently our handling for rocprofiler operations are explicitly synchronized by us so that the application thread that launches a kernel will be blocked until we have processed the rocprofiler data for the kernel. So at this point, there is no need for additional flush.